### PR TITLE
feat: update to rxjs 5.5.0 and switch to lettable operators

### LINKED ISSUE #7275: feat(rxjs): update to rxjs@5.5.0
#### Bug, feature request, or proposal:

Feature request

#### What is the expected behavior?

Should adopt the newest version of RxJS, with support for lettable operators:
`import {first} from 'rxjs/operator/first';` -> `import {first} from 'rxjs/operators';`

#### What is the current behavior?

Currently uses a homebrew version of lettable imports

#### What is the use-case or motivation for changing an existing behavior?

The newest version is clearer and consolidates the source of imports. Also, angular/angular is moving to it with https://github.com/angular/angular/pull/19345

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

angular 5.0.0-beta.7
material 2.0.0-beta.11

### PR DESCRIPTION
**Pull request by crisbeto**
Bumps the required RxJS version to 5.5.0 and gets rid of our RxChain in favor of using the lettable Rx operators. Refactors all the usages and updates the configs.

Fixes #7275.

### CODE REVIEW

**Comment by jelbourn** [comment]
Also blocked on rxjs 5.5 getting into google

**Comment by CaerusKaru** [comment]
Is this unblocked now that https://github.com/angular/angular/pull/19345 is synced? Or are we waiting for 5.0.0 final?

File: package.json [hunk 1]
```diff
@@ -32,8 +32,9 @@
     "@angular/forms": "~4.4.4",
     "@angular/http": "~4.4.4",
     "@angular/platform-browser": "~4.4.4",
+    "a-sync-waterfall": "^1.0.0",
     "core-js": "^2.4.1",
-    "rxjs": "^5.0.1",
+    "rxjs": "5.5.0-beta.7",
```
Comment (CaerusKaru) [inline package.json]: RxJS 5.5.0 is now out of beta, this can be set to `^5.5.0`

**Comment by crisbeto** [comment]
Rebased, sorted out any new usages of the `RxChain` and bumped to 5.5.0 final. @jelbourn unblocking this one and setting it as P1 since it might be nice to have in the RC.

**Comment by IgorMinar** [comment]
Awesome work!

**Pull request merged**